libs/entry-components/configure-theming.md

```diff
@@ -0,0 +1,435 @@
+
+- [`entry-components`](https://www.npmjs.com/package/@enigmatry/entry-components)
+- [`entry-form`](https://www.npmjs.com/package/@enigmatry/entry-form)
+
+## Project structure
+
+We require a dedicated file to include theme-related style configurations. Adhering to our [SCSS coding standard](https://wiki.enigmatry.com/en/standards-and-guidelines/sass-coding-standard), this file should be within the `styles/partials/vendors/libraries` subfolder. If the `entry` subfolder is not already defined in the project, create it. Within the `entry` subfolder, the following SCSS partials should be created:
+
+```diff
+├── libraries
++    └── entry
++      ├── _general.scss
++      ├── _scss-foundation.scss
++      └── _index.scss
+```
+
+- `general.scss` -Imports necessary library generator files for configuring theming.
+- `scss-foundation.scss` - imports the scss foundation library
+- `index.scss` entry point for maintaining good SCSS structure
+
+## Styles Configuration
+
+Within the `general.scss` partial, import the generator file from the library and our custom variables related to the project.
+
+The generator file represents a collection of includes and a mixin that, when used, generates styles for various components based on a combination of a default theme and a custom theme. The generators for each entry component handle the specifics of styling based on the provided theme. You can check which components are supported on the [Components demo](https://entry-demo.enigmatry.com/) application.
+
+In summary, the generator:
+
+- Includes multiple generator files for supported entry components (dialog, table, button, form...)
+- Generates [Angular Material](https://material.angular.io/guide/theming) for customization of color pallets, typography, and density settings
+- Sets up  default, boilerplate theme with predefined values
+- Combines default theme and custom theme styles
+
+```scss
+@use 'entry-components/styles/generator';
+```
+
+## Create custom theme
+
+Next, define the $custom-theme map that will be expanded with various customization options based on specific needs.
+
+```scss
+$custom-theme: ();
+```
+
+Include and apply the styles generated by the generate-from mixin in the 'generator' module, using the customization options provided in the $custom-theme map.
+
+```scss
+@include generator.generate-from($custom-theme);
+```
+
+Our theme is now an empty object, and default library theme is used.
+
+```scss
+@use 'entry-components/styles/generator';
+
+$custom-theme: (
+);
+
+@include generator.generate-from($custom-theme);
+
+```
+
+## Importing Fonts
+
+There are 2 possibilities of using fonts in theming:
+
+### 1. Use default fonts from library
+
+**Entry-components** library comes with default fonts:  **Montserrat**, **OpenSans** and **Roboto** font family. To be able to use them, it needs to be configured in the `angular.json` file located at the root of the app. The fonts are imported from the library's node modules as follows:  
```

Comment:
Only Roboto is used. Other fonts are commented out for time being.

Reply:
Montserrat and Open Sans are technically still located in the assets, but they are commented out and not in use. Therefore, I will remove them from the documentation.